### PR TITLE
Move MacAddr to utils

### LINKED ISSUE #1765: move `MacAddr` from dumbo to utils
`MacAddr` is a simple wrapper and a utility used by multiple crates. Right now this is in `dumbo` and `dumbo` is used in a dependency in the following crates:
- vmm for `MacAddr`
- devices for `MacAddr` and for the MMDS network stack
- utils for `MacAddr`

By moving `MacAddr` to `utils`, we make `utils` a tinier crate. We can also push this code to rust-vmm/vmm-sy-util.

### PR DESCRIPTION
**Pull request by jmcconnell26**
## Reason for This PR

Move MacAddr from dumbo crate to net
Fixes #1765.

## Description of Changes

Move MacAddr from dumbo crate to net. Update references to reflect change. Remove now unneeded dependency on dumbo from devices.

- [ ] This functionality can be added in [`rust-vmm`](https://github.com/rust-vmm).

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license.

## PR Checklist

`[Reviewer TODO: Verify that these criteria are met. Request changes if not]`

- [X] All commits in this PR are signed (`git commit -s`).
- [X] The reason for this PR is clearly provided (issue no. or explanation).
- [X] The description of changes is clear and encompassing.
- [X] Any required documentation changes (code and docs) are included in this PR.
- [X] Any newly added `unsafe` code is properly documented.
- [X] Any API changes are reflected in `firecracker/swagger.yaml`.
- [X] Any user-facing changes are mentioned in `CHANGELOG.md`.
- [X] All added/changed functionality is tested.

### CODE REVIEW

File: src/utils/Cargo.toml [hunk 1]
```diff
@@ -6,6 +6,11 @@ edition = "2018"
...
 [dependencies]
 libc = ">=0.2.39"
+serde = ">=1.0.27"
 vmm-sys-util = ">=0.6.1"
 
 net_gen = { path = "../net_gen" }
+
+
+[dev-dependencies]
+serde_json = ">=1.0.9"
```
Comment (iulianbarbu) [inline src/utils/Cargo.toml:16]: nit: Please add a newline.

**Comment by iulianbarbu** [comment]
Hi, @jmcconnell26! Thanks for the PR.
Can you please squash all the commits into a single commit? 
The commit must respect the 50/72 commit rule and it must be signed as in `git commit -s --amend`.

**Comment by jmcconnell26** [comment]
@iulianbarbu, thanks for the review!
I have updated the git history and made the requested change.
I'm not sure if the form is to put PR requested changes into their own commits or not, please let me know if you would like me to change this.

**Comment by iulianbarbu** [comment]
> @iulianbarbu, thanks for the review!
> I have updated the git history and made the requested change.
> I'm not sure if the form is to put PR requested changes into their own commits or not, please let me know if you would like me to change this.

Please put the PR requested changes also in the first commit.

**Comment by dianpopa** [comment]
* This line can also be removed now: https://github.com/jmcconnell26/firecracker/blob/ISSUE-1765-MoveMacAddrToUtils/src/vmm/Cargo.toml#L20.

* I suggest modifying the commit title to: "dumbo: move mac module to utils"
At the end of the  commit body you can write: "Fixes #1765" which will trigger the closing of the issue once the PR is merged.

**Comment by jmcconnell26** [comment]
@iulianbarbu, @dianpopa, thanks both for taking the time to review!
I have removed the dependency on dumbo, as well as updating the commit message.
If there are are any more changes you would like me to make, please let me know.

**Review by dianpopa — APPROVED** [review]

**Review by iulianbarbu — APPROVED** [review]

**Pull request merged**